MapboxCoreNavigation/RouteController.swift

```diff
@@ -262,6 +262,20 @@ open class RouteController: NSObject {
      */
     var rawLocation: CLLocation?
     
+    public var reroutingTolerance: CLLocationDistance {
+        guard let intersections = routeProgress.currentLegProgress.currentStepProgress.intersectionsIncludingUpcomingManeuverIntersection else { return RouteControllerMaximumDistanceBeforeRecalculating }
+        guard let userLocation = rawLocation else { return RouteControllerMaximumDistanceBeforeRecalculating }
+        
+        for intersection in intersections {
+            let distanceToIntersection = distance(along: routeProgress.currentLegProgress.currentStepProgress.step.coordinates!, from: userLocation.coordinate, to: intersection.location)
+            
+            if distanceToIntersection <= RouteControllerManeuverZoneRadius {
```

Comment:
@1ec5 I moved away from keeping count of the current intersection. This is because this line of code is hit right when the user is reaching the intersection. If we increment the intersection index here, we'd already be looking at the next intersection while the user is still in the current intersection.

We could alternatively check the users course compared to the expected outlet heading, but then again, straight intersections will be difficult.

Noting this for loop should remain light as there are never really that many intersections on a step. _Waits patient to be shown wrong._

Reply:
[108 intersections](https://www.mapbox.com/get-directions/#16.18/39.73485/-105.1595?coordinates=-105.159523,39.734874;-105.022986,39.740455) over a 7½-mile stretch of [Colfax Avenue](https://en.wikipedia.org/wiki/Colfax_Avenue) in Denver. Examples like these would be more common once we implement toll and freeway avoidance.